### DISCUSSION TITLE
New feature! Individual scrobble pages and embeddable

### DISCUSSION THREAD

**Discussion by mattl** [foocorp/hacienda#90] [Future enhancements]
A discussion came up tonight on an IndieWebCamp Homebrew Website Club meeting around embedding.

You can now get a unique URL for each track you scrobble (retroactively too) plus you can embed the page with very minimal HTML into say, a blog post.

1. Go to your profile (ie. https://libre.fm/user/mattl)
2. Click on the text in the "time" column of the recent tracks list
3. https://libre.fm/user/mattl/scrobble/1728341197

You can also use https://libre.fm/user/mattl/scrobble/1728341197/embed to get the embedded version.

Let me know what you think and @foocorp/beta-testers let me know if you find any bugs.

**Comment by mattl** [comment]
Some thoughts:

* ~Could probably display the time/date in a more useless manner.~
* ~Show a profile icon for the user~
* Show album art if we have it (we probably don't right now)
* ~add h-entry, etc to the non-embed version~
* are we doing the best job of markup?

Anyway for a first draft this just added another 600 million+ new URLs to the web :)

**Comment by pluc** [comment]
A cool idea could be to add song data (eg scrobbled X times or scrobbled by X users), third-party embeds (youtube/spotify/whatever), lyrics or whatever else that's part of the ecosystem that we want to bring visibility to

**Comment by unseenlarks** [comment]
would it be possible to delete a scrobble from its individual page, and hence the page itself?

**Comment by mattl** [comment]
@unseenlarks yes, I think that'd be possible. 

@pluc The idea is that it's more like a tweet/toot/post/skeet/status update.

**Comment by mattl** [comment]
@unseenlarks I started putting this idea together tonight.

<img width="621" alt="Screenshot 2024-10-24 at 5 50 44 PM" src="https://github.com/user-attachments/assets/62736174-aa49-46fd-8bea-c4c7afe40039">

In the first screenshot, this is a scrobble by Evan Prodromou, not me.. so I cannot delete it.

<img width="622" alt="Screenshot 2024-10-24 at 5 50 55 PM" src="https://github.com/user-attachments/assets/1becb9af-300c-4683-9a90-d5abb0d19c42">

In the second screenshot, this is a scrobble I made so I should be able to delete it.

**Comment by mattl** [comment]
Happy with this for now.

**Comment by mattl** [comment]
Reopening this one. 

Plain text and image versions were added today.

Here’s how it works:

You have a scrobble, in my case:

https://libre.fm/user/mattl/scrobble/1742439265

There’s already the embed version which is minimal HTML

https://libre.fm/user/mattl/scrobble/1742439265/embed

Today I added plain text and images:

* Plain text: https://libre.fm/user/mattl/scrobble/1742439265/txt

* Image: https://libre.fm/user/mattl/scrobble/1742439265/image

There’s also example HTML and BBCode (for web based forums, etc) provided. I may need a second example of BBCode as it seems the alt text doesn’t work on every forum. 

I’m looking for a new font for the image version that will handle non-Roman characters. @kevinmarks suggested Noto, so I’ll try that tomorrow.

**Comment by sweiglbosker** [comment]
Is there an easy way to do this but for the most recent scrobble?
